pkg/sql/compile/scope.go

```diff
@@ -187,10 +188,37 @@ func (s *Scope) ParallelRun(c *Compile, remote bool) error {
 		}
 		s.NodeInfo.Data = nil
 	case s.NodeInfo.Rel != nil:
-		var err error
-
-		if rds, err = s.NodeInfo.Rel.NewReader(c.ctx, mcpu, s.DataSource.Expr, s.NodeInfo.Data); err != nil {
+		// get readers of main table
```

Comment:
For partitioned tables, rel should have to be nil. please treat it this way. nodeInfo.Rel is a weird thing. None of it should go here anymore, except to ensure historical compatibility.

Reply:
OK, I will deal with the partition table. The tableScan of the partition table does not go through case s.NodeInfo.Rel= Nil: Branch